eclipse-theia/theia
# Apply FileDecoration provider API

#### What it does

Align the FileDecoration vscode Plugin API with the latest vscode version. This API was changed and it is used in the vscode git plugin, so to be able to use the latest vscode git plugin it is needed to apply the changes in the Theia's FileDecoration plugin API.

related CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22921
#### How to test


1. Remove `"@theia/git": "^1.9.0"` from `examples/browser/package.json` file.
2. Download [vscode-git-1.52.1](https://open-vsx.org/api/vscode/git/1.52.1/file/vscode.git-1.52.1.vsix) and copy it to the `plugins` folder.
3. Start Theia and  open a project cloned from GitHub.
4. Make some changes in the project and check them in the SCM panel:
![screenshot-localhost_3000-2020 12 29-17_06_55](https://user-images.githubusercontent.com/7668752/103293267-3253c580-49f8-11eb-8d43-305433e8537d.png)



#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
